corelib/src/cheatcodes.cairo

```diff
@@ -118,3 +118,27 @@ fn stop_warp(contract_address: ContractAddress) {
     cheatcode::<'stop_warp'>(array![contract_address_felt].span());
 }
 
+fn start_mock_call(contract_address: ContractAddress, fn_name: felt252, ret_data: Array::<felt252>) {
```

Comment:
```suggestion
fn start_mock_call<T>(contract_address: ContractAddress, fn_name: felt252, ret_data: T) {
```

I'm not sure if something like that is possible but definitely would be helpful

Reply:
Okay, I was wrong, it is definitely possible. An example from my PR

```cairo
trait TxtParser<T, impl TSerde: Serde<T>> {
    fn deserialize_txt(file: @File) -> Option<T>;
}

impl TxtParserImpl<T, impl TSerde: Serde<T>> of TxtParser<T> {
    fn deserialize_txt(file: @File) -> Option<T> {
        let mut content = cheatcode::<'parse_txt'>(array![*file.path].span());
        Serde::<T>::deserialize(ref content)
    }
}
```

Reply:
Added in https://github.com/foundry-rs/starknet-foundry/pull/385/commits/1bade0618e763faa41fd53c94a198010e10d7f3a